common/source/docs/common-dock-mode.rst

```diff
@@ -0,0 +1,103 @@
+.. _common-dock-mode:
+
+=========
+Dock Mode
+=========
+
+The DOCK mode maneuvers a vehicle towards a stationary docking target automatically.
+Currently, we support docking with the help of a camera attached to a companion computer.
+The companion computer estimates the pose of the marker (ArucoMarker/AprilTag) placed at the docking station and sends
+`LANDING_TARGET <https://mavlink.io/en/messages/common.html#LANDING_TARGET>`__ mavlink messages to the vehicle which are then processed by the precision landing library
+(`AC_PrecLand <https://github.com/ArduPilot/ardupilot/tree/master/libraries/AC_PrecLand>`__) to estimate the position of the docking target.
+
+Successful tests have been conducted using the OpenMV camera as a companion computer to perform auto-docking.
+However, any companion computer equipped with an RGB camera (like OpenMV camera, Raspberry Pi, Jetson Nano etc.) which can estimate pose
+of marker placed at docking target and send mavlink messages to the vehicle can be used.
+
+The OpenMV Camera
+=================
+
+OpenMV camera is a pocket sized camera board which is capable of performing various computer vision tasks.
+It can transmit mavlink messages over UART interface and hence it can be used as a companion computer to perform auto-docking.
+The camera can be purchased directly from `openmv.io <https://openmv.io/collections/cams>`__
+
+Connection to Autopilot
+=======================
+
+The companion computer needs to be connected to the autopilot through UART interface.
+
+.. image:: ../../../images/openmv-pixhawk.jpg
+   :target: ../_images/openmv-pixhawk.jpg
+   :width: 600px
+
+- Connect the camera’s P4, VIN and GND pins to one of the autopilot’s serial ports as shown above. In the image above the OpenMV cam is connected to a Pixhawk’s Telem2 port.
+  The connection scheme would be same for the other alternatives too. The user just needs to make a connection between
+  TX pin on the companion computer to RX pin of the telemetry port on the autopilot and power up the autopilot.
+- Set :ref:`SERIAL2_PROTOCOL <SERIAL2_PROTOCOL>` = 1 (MAVLink) if using Serial2/Telem2, if connected to another serial port use the corresponding SERIALx_PROTOCOL parameter
+- Set :ref:`SERIAL2_BAUD <SERIAL2_BAUD>` = 115
+
+Generating AprilTag Markers
+===========================
+
+.. image:: ../../../images/openmv-apriltag-generator.png
+   :target: ../_images/openmv-apriltag-generator.png
+   :width: 600px
+
+The OpenMV camera has capability to detect the AprilTags. The OpenMV IDE comes with built-in AprilTag generator. The steps to generate an AprilTag using the same are:
+
+- Go to Tools -> Machine Vision -> AprilTag generator -> TAG36H11 Family (587 Tags - Recommended). This will open a small dialog box.
+- Set the range of tags you want to generate. For eg. if you want to generate tags numbered 1 to 5, set Min to 1 and Max to 5.
+- Click ok and save the tags to the desired directory.
+
```

Comment:
now what? print it out? how big?and how should it be placed at the docking location?you should reference the the script and the script's structure for the tag numbering and border size....its not clear immediately from the script's comments that its of the format tagid:border length in mm

Reply:
Thanks for letting me know that I missed it. It is basically users decision how bigger marker he wants to use. The default as per the script is 165 mm (which is got when we simply print the generated image on an A4 sheet). However, a bigger marker can be used to increase the range.